WalletWasabi/Services/WalletService.cs

```diff
@@ -586,41 +586,52 @@ public Node LocalBitcoinCoreNode
 			private set => _localBitcoinCoreNode = value;
 		}
 
+		/// <param name="hash">Block hash of the desired block, represented as a 256 bit integer.</param>
 		/// <exception cref="OperationCanceledException"></exception>
-		public async Task<Block> GetOrDownloadBlockAsync(uint256 hash, CancellationToken cancel)
+		public async Task<Block> FetchBlockAsync(uint256 hash, CancellationToken cancel)
+		{
+			Block block = await TryGetBlockFromFileAsync(hash, cancel);
+			if (block is null)
+			{
+				block = await DownloadBlockAsync(hash, cancel);
+			}
+			return block;
+		}
+
+		/// <param name="hash">Block hash of the desired block, represented as a 256 bit integer.</param>
+		/// <exception cref="OperationCanceledException"></exception>
+		private async Task<Block> TryGetBlockFromFileAsync(uint256 hash, CancellationToken cancel)
```

Comment:
If the method's name is `TryXXX` then it should follow the pattern:

```c#
private async Task<bool> TryGetBlockFromFileAsync(uint256 hash, out Block block, CancellationToken cancel)
```

Reply:
@lontivero No. There's no `out` in async methods, in this case the return value is null if failure.